eox_nelp/course_experience/api/v1/tests/mixins_helpers.py

```diff
@@ -0,0 +1,333 @@
+"""This file contains mixins to use for experience views test cases.
+Mixins:
+    - ExperienceTestMixin.
+    - UnitExperienceTestMixin
+    - CourseExperienceTestMixin
+"""
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from eox_nelp.course_experience.api.v1.views import INVALID_KEY_ERROR
+from eox_nelp.edxapp_wrapper.course_overviews import CourseOverview
+
+User = get_user_model()
+
+RESPONSE_CONTENT_TYPES = ["application/vnd.api+json", "application/json"]
+BASE_ITEM_ID = "block-v1:edX+cd1011+2020t1+type@vertical+block@base_item"
+BASE_COURSE_ID = "course-v1:edX+cd101+2023-t2"
+
+
+class ExperienceTestMixin:
+    """
+    A mixin class with base experience setup configuration for testing.
+    """
+    def setUp(self):  # pylint: disable=invalid-name
+        """
+        Set base variables and objects across experience test cases.
+        """
+        self.client = APIClient()
+        self.user, _ = User.objects.get_or_create(username="vader")
+        self.my_course, _ = CourseOverview.objects.get_or_create(id=BASE_COURSE_ID)
+        self.client.force_authenticate(self.user)
+
+    def test_get_object_list_by_user(self):
```

Comment:
how do you ensure that the user has access to the returned data I mean there is a modification in the method get_queryset that filters the result, what is the test that verifies that the returned data is filtered

Reply:
for the querydict I added a test for forms.
[fc6decf](https://github.com/eduNEXT/eox-nelp/pull/34/commits/fc6decf7d73423604061f8f235b1461258aa6e69)
For the user I check it:
[d08f0fd](https://github.com/eduNEXT/eox-nelp/pull/34/commits/d08f0fd4ed74331d6dc6e9b869ea48fc8ebafefd)